### ISSUE TITLE
Use precision-recall curve to identify optimum classification thresholds

### ISSUE THREAD

**Issue by yiwen-h** [CDU-data-science-team/pxtextmining#127]
- Trusts tend to prioritise recall over precision
- What is an acceptable recall/precision rate?
- Different thresholds for different categories?

**Comment by yiwen-h** [comment]
- In order to do this we need to create a validation set (prevent overfitting to test set). We already do this for the BERT model so easy enough to do with the sklearn pipeline.
- Use `sklearn.metrics.precision_recall_curve` - treat it like a binary classification problem (one vs rest) to get the threshold for optimising F1 score for each class.
